### ISSUE TITLE
Consul service discovery filters

### ISSUE THREAD

**Issue by f41gh7** [VictoriaMetrics/VictoriaMetrics#4183]
### Is your feature request related to a problem? Please describe

 Since v1.14, consul supports new filtering features for API requests https://github.com/hashicorp/consul/issues/9422

https://developer.hashicorp.com/consul/api-docs/features/filtering

 It implements advanced server-side filter which reduce amount of traffic between consul server and discovery client.

### Describe the solution you'd like

 Introduce new config option - `filter`.

```yaml
# promscrape.yaml
scrape_configs:
  - job_name: consul-targets
     consul_sd_configs:
     - filter:  'dc-1 in ServiceTags or dc-2 in ServiceTags and NodeMeta Is Not Empty' 
```

### Describe alternatives you've considered

 Use [relabel_configs](https://docs.victoriametrics.com/relabeling.html) for filtering, but it's not flexible and consumes additional resources.

**Comment by valyala** [comment]
The related issue - https://github.com/VictoriaMetrics/VictoriaMetrics/issues/574

**Comment by valyala** [comment]
The `filter` option is available in `vmagent` starting from the commit 5ee344824ffe6509a5caaa3ddcdb0ccbd7a1de0d . This commit will be included in the next release. In the mean time it is possible to build vmagent from this commit according to [these docs](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verify whether the `filter` option works as expected.

**Comment by valyala** [comment]
The `filter` option is available at [consul_sd_configs](https://docs.victoriametrics.com/sd_configs.html#consul_sd_configs) starting from [v1.91.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the feature request as done.
